server/tests/api/videos/video-privacy.ts

```diff
@@ -164,6 +175,45 @@ describe('Test video privacy', function () {
     }
   })
 
+  it('Should upload a non-federating unlisted video to server 2', async function () {
+    this.timeout(30000)
+
+    customConfig.federation.videos.federateUnlisted = false
+    await updateCustomConfig(servers[1].url, servers[1].accessToken, customConfig)
```

Comment:
If the setting cannot be updated using the web interface, take example on https://github.com/Chocobozzz/PeerTube/blob/develop/server/tests/api/redundancy/redundancy-constraints.ts#L124 to update the server config